File: BAN6420 Module 1 assignment.py
#BAN6420 Mosdule 1 assignment
import random

# Generating a list of workers dynamically
num_workers = 451
workers = []
for i in range(num_workers):
    worker = {
        'name': f'Staff_102{i+1}',
        'salary': random.randint(5000, 35000),  # Generating random salary between $5000 and $35000
        'gender': random.choice(['Male', 'Female'])
    }
    workers.append(worker)

# Function to generate payment slips
def generate_payment_slips(workers):
    payment_slips = []
    for worker in workers:
        try:
            slip = {
                'name': worker['name'],
                'salary': worker['salary'],
                'gender': worker['gender'],
                'employee_level': ''
            }
            if 10000 < worker['salary'] < 20000:
                slip['employee_level'] = 'A1'
            if 7500 < worker['salary'] < 30000 and worker['gender'] == 'Female':
                slip['employee_level'] = 'A5-F'
            payment_slips.append(slip)
        except KeyError as e:
            print(f"Missing key: {e}")
        except Exception as e:
            print(f"Error: {e}")
    return payment_slips

# Generate payment slips for all workers
payment_slips = generate_payment_slips(workers)
print(f"Generating Payslips...{len(payment_slips)} payment slips\n")
print("Done!\n")
# Print a few example payment slips
numpaymentslips = None
while type(numpaymentslips) != int or  numpaymentslips <= 0 or numpaymentslips > len(payment_slips) :
    try:
        numpaymentslips = int(input("How many payment slips do you want? "))  # Ask user how many payment slips they want
        if numpaymentslips > len(payment_slips) or numpaymentslips <= 0:
            raise Exception
    except Exception as e:
            print(f"Please enter a number between 0 and {len(payment_slips)}")
    except ValueError:
        print("Please enter a valid number.")
    else:
        for slip in payment_slips[:numpaymentslips]:
            print(slip)

File: BAN6420 Module 1 assignment.r
# Load required library
library(dplyr)

# Generate a list of workers dynamically
set.seed(123)  # For reproducibility
num_workers <- 451
workers <- data.frame(
  name = paste0("Staff_102", 1:num_workers),
  salary = sample(5000:35000, num_workers, replace = TRUE),
  gender = sample(c("Male", "Female"), num_workers, replace = TRUE)
)

# Function to generate payment slips
generate_payment_slips <- function(workers) {
  workers <- workers %>%
    mutate(employee_level = case_when(
      salary > 10000 & salary < 20000 ~ "A1",
      salary > 7500 & salary < 30000 & gender == "Female" ~ "A5-F",
      TRUE ~ ""
    ))
  return(workers)
}

# Generate payment slips for all workers
payment_slips <- generate_payment_slips(workers)

# Print a few example payment slips
print(paste("Generating Payslips...", nrow(payment_slips), "payment slips"))
print("Done!\n")

# Function to print specific number of payment slips based on user input
print_payment_slips <- function(payment_slips) {
  numpaymentslips <- NA
  while(is.na(numpaymentslips) || numpaymentslips <= 0 || numpaymentslips > nrow(payment_slips)) {
    numpaymentslips <- as.numeric(readline(paste("How many payment slips do you want? (1 -", nrow(payment_slips), "): ")))
    if (is.na(numpaymentslips) || numpaymentslips <= 0 || numpaymentslips > nrow(payment_slips)) {
      cat("Please enter a number between 1 and", nrow(payment_slips), "\n")
    }
  }
  print(head(payment_slips, numpaymentslips))
}

# Ask user for the number of payment slips to print
print_payment_slips(payment_slips)
